be/src/olap/rowset/beta_rowset_reader.cpp

```diff
@@ -59,27 +59,46 @@ Status BetaRowsetReader::init(RowsetReaderContext* read_context) {
                                                  read_context->is_upper_keys_included->at(i));
         }
     }
+
+    bool can_reuse_schema = true;
     // delete_hanlder is always set, but it maybe not init, so that it will return empty conditions
     // or predicates when it is not inited.
     if (read_context->delete_handler != nullptr) {
         read_context->delete_handler->get_delete_conditions_after_version(
                 _rowset->end_version(), &read_options.delete_conditions,
                 read_options.delete_condition_predicates.get());
+        // if del cond is not empty, schema may be different in multiple rowset
+        can_reuse_schema = read_context->delete_handler->empty();
```

Comment:
Why does we consider delete hander here? It seems that we use only one schema for a tuple reader even there are delete handler.

Reply:
In function `get_delete_conditions_after_version`, it will get delete_condition_predicates according to rowset version. it will lead that schema may be not the same.
```
        read_context->delete_handler->get_delete_conditions_after_version(
                _rowset->end_version(), &read_options.delete_conditions,
                read_options.delete_condition_predicates.get());
        ....
        ....
        read_options.delete_condition_predicates->get_all_column_ids(delete_columns_set);
        for (auto cid : delete_columns_set) {
            if (read_columns_set.find(cid) == read_columns_set.end()) {
                read_columns.push_back(cid);
            }
        }
        _input_schema = std::make_shared<Schema>(_context->tablet_schema->columns(), read_columns);

```